custom_metrics/privacy.js

```diff
@@ -15,10 +46,264 @@ return JSON.stringify({
+    let ats = {
+      present: meta_tag != null,
+      ads_metadata: null,
+    };
+    if (ats.present) {
+      ats.ads_metadata = meta_tag.content;
+    }
+    return ats;
+  })(),
+
+  /**
+   * FLoC
+   *
+   * Test site: https://floc.glitch.me/
+   * Test site: https://www.pokellector.com/
+   *
+   * @todo Check function/variable accesses through string searches (wrappers cannot be used, as the metrics are only collected at the end of the test)
+   */
+  floc: {
+    document_interestCohort:
+      findStringInBodies('document.interestCohort') ||
+      findStringInBodies('interestCohort'),
+  },
+
+  /**
+   * Do Not Track (DNT)
+   * https://www.eff.org/issues/do-not-track
+   *
+   * Test site: https://www.theverge.com/
+   */
+  navigator_doNotTrack:
+    findStringInBodies('navigator.doNotTrack') ||
+    findStringInBodies('doNotTrack'),
+
+  /**
+   * Global Privacy Control
+   * https://globalprivacycontrol.org/
+   *
+   * Test site: https://global-privacy-control.glitch.me/
+   */
+  navigator_globalPrivacyControl:
+    findStringInBodies('navigator.globalPrivacyControl') ||
+    findStringInBodies('globalPrivacyControl'),
+
+  // Sensitive resources
+
+  /**
+   * Permissions policy
+   * https://www.w3.org/TR/permissions-policy-1/#introspection
+   */
+  document_permissionsPolicy:
+    findStringInBodies('document.permissionsPolicy') ||
+    findStringInBodies('permissionsPolicy'),
+
+  /**
+   * Feature policy 
+   * (previous name of Permission policy: https://www.w3.org/TR/permissions-policy-1/#introduction)
+   */
+  document_featurePolicy:
+    findStringInBodies('document.featurePolicy') ||
+    findStringInBodies('featurePolicy'),
+
+  // Permissions Policy / Feature Policy on iframes already implemented in `security.js` custom metrics.
+
+  /**
+   * Referrer Policy
+   * https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Referrer-Policy
+   */
+  referrerPolicy: (() => {
+    let rp = {
+      entire_document_policy: null,
+      individual_requests: null,
+      link_relations: null
+    };
+    // Referrer policy set for entire document using `meta` tag
+    // Test site: https://www.cnet.com/
+    let referrer_meta_tag = document.querySelector(
+      'meta[name="referrer"]'
+    );
+    if (referrer_meta_tag) {
+      rp.entire_document_policy = referrer_meta_tag.content; // Get policy value
+    }
+    // Referrer policy set for individual requests with the `referrerpolicy` attribute
+    // Test site: https://www.brilio.net/
+    let referrerpolicy_attributes = document.querySelectorAll(
+      '[referrerpolicy]'
+    );
+    // Leave `individual_requests` at `null` if no attributes are found.
+    if (referrerpolicy_attributes.length > 0) {
+    // Build dictionary of occurrences of tag-value pairs.
+      rp.individual_requests = Object.fromEntries(
+        Array.from(referrerpolicy_attributes).map(
+          x => (x.tagName + '|' + x.getAttribute('referrerpolicy'))
```

Comment:
WDYT about adding more structure rather than delimiting by `'|'`? For example, you could have an object like `{tagName: x.tagName, referrerpolicy: x.getAttribute('referrerpolicy'), count: 1}` and the `reduce` method could increment the `count` property.

A structure like that is more extensible if needed and may facilitate aggregation/analysis later.

Reply:
I implemented an approach similar to yours in cee66cb, I don't know if that is what you had in mind.
(I wanted to store it as an object originally as well, but didn't think further than to use a hash map, which fails because objects with the same properties are not equal.)